text/trace/0173-messaging-semantic-conventions.md

```diff
@@ -0,0 +1,255 @@
+                                   v                                
++---------+ +---------+ +---------+ +---------+ +---------+ +---------+
+|Message A| |Message B| |Message C| |Message D| |Message E| |Message F|
++---------+ +---------+ +---------+ +---------+ +---------+ +---------+
+                     <---  Settled
+```
+
+#### Examples
+
+1. The following configurations should be instrumented and tested for Kafka or
+   a similar messaging system:
+
+   * 1 producer, 2 consumers in the same consumer group
+   * 1 producer, 2 consumers in different consumer groups
+   * 2 producers, 2 consumers in the same consumer group
+
+   Each of the producers produces a continuous stream of messages.
+
+## Open questions
+
+The following areas are considered out-of-scope of a first stable release of
+semantic conventions for messaging. While not being explicitly considered for
+a first stable release, it is important to ensure that this first stable
+release can serve as a solid foundation for further improvements in these areas.
+
+### Sampling
+
+The current experimental semantic conventions rely heavily on span links as
+a way to correlate spans. This is necessary, as several traces are needed to
+model the complete path that a message takes through the system. With the currently
+available sampling capabilities of OpenTelemetry, it is not possible to ensure
+that a set of linked traces is sampled. As a result, it is unlikely to sample a
+set of traces that covers the complete path a message takes.
+
+Solving this problem requires a solution for sampling based on span links,
+which is not in scope for this OTEP.
+
+However, having a too high number of span links in a single trace or having too
+many traces linked together can make the visualization and analysis of traces
+inefficient. This problem is not related to sampling and needs to be addressed
+by the semantic conventions.
+
+### Instrumenting intermediaries
+
+Instrumenting intermediaries can be valuable for debugging configuration or
+performance issues, or for detecting specific intermediary failures.
+
+Stable semantic conventions for instrumenting intermediaries can be provided at
+a future point in time, but are not in scope for this OTEP. The messaging
+semantic conventions this document refers to need to provide instrumentation
+that works well without the need to have intermediaries instrumented.
+
+### Metrics
+
+Messaging semantic conventions for tracing and for metrics overlap and should
+be as consistent as possible. However, semantic conventions for metrics will be
+handled separately and are not in scope for this OTEP.
+
+### Asynchronous message passing in the wider sense
+
+Asynchronous message passing in the wider sense is a communication method
+wherein the system puts a message in a queue or channel and does not require an
+immediate response to continue processing. This can range from utilizing a
+simple queue implementation to a full-fledged messaging system.
```

Comment:
Please add some examples of what kind of systems/scenarios are meant here so it's clearer which particular frameworks are left out of scope for the messaging conventions. The description of https://github.com/open-telemetry/opentelemetry-specification/pull/1582 might be a good reference there.

Reply:
I added some examples, and a reference to the PR.